src/push-state-tree.js

```diff
@@ -106,49 +156,63 @@ var elementPrototype = typeof HTMLElement !== 'undefined' ? HTMLElement : Elemen
 
 function PushStateTree(options) {
   options = options || {};
-  options[USE_PUSH_STATE] = options[USE_PUSH_STATE] !== false;
 
   // Force the instance to always return a HTMLElement
   if (!(this instanceof elementPrototype)) {
     return PushStateTree.apply(PushStateTree.createElement('pushstatetree-route'), arguments);
   }
 
-  this.VERSION = VERSION;
+  this.eventStack = {
+    leave: [],
+    change: [],
+    enter: [],
+    match: []
+  };
 
-  // Setup options
-  for (var prop in options) {
-    if (options.hasOwnProperty(prop)) {
-      this[prop] = options[prop];
+  //TODO: emcapsulate this
+  for (let property in PushStateTree.prototype) {
+    if (typeof PushStateTree.prototype[property] === 'function') {
+      // function wrapper, without bind the context
+      this[property] = PushStateTree.prototype[property].bind(this);
+      continue;
+    }
+    // Copy properties from prototype to the instance
+    if (typeof this[property] == 'undefined') {
+      this[property] = PushStateTree.prototype[property];
     }
   }
```

Comment:
While I understand that it's nice to not worry about the bound methods, is this really necessary?